src/System.IO.Packaging/src/System/IO/Packaging/ContentType.cs

```diff
@@ -0,0 +1,744 @@
+        private static string ValidateToken(string token)
+        {
+            if (String.CompareOrdinal(token, String.Empty) == 0)
+                throw new ArgumentException(SR.Get(SRID.InvalidToken));
+
+            for (int i = 0; i < token.Length; i++)
+            {
+                if (IsAsciiLetterOrDigit(token[i]))
+                    continue;
+                else
+                    if (IsAllowedCharacter(token[i]))
+                    continue;
+                else
+                    throw new ArgumentException(SR.Get(SRID.InvalidToken));
+            }
+
+            return token;
+        }
+
+        /// <summary>
+        /// Validating if the value of a parameter is either a valid token or a 
+        /// valid quoted string
+        /// </summary>
+        /// <param name="parameterValue">paramter value string</param>
+        /// <returns>validate parameter value string</returns>
+        /// <exception cref="ArgumentException">If the paramter value is empty</exception>
+        private static string ValidateQuotedStringOrToken(string parameterValue)
+        {
+            if (String.CompareOrdinal(parameterValue, String.Empty) == 0)
+                throw new ArgumentException(SR.Get(SRID.InvalidParameterValue));
+
+            if (parameterValue.Length >= 2 &&
+                parameterValue.StartsWith(_quote, StringComparison.Ordinal) &&
+                parameterValue.EndsWith(_quote, StringComparison.Ordinal))
+                ValidateQuotedText(parameterValue.Substring(1, parameterValue.Length - 2));
+            else
+                ValidateToken(parameterValue);
+
+            return parameterValue;
+        }
+
+        /// <summary>
+        /// This method validates if the text in the quoted string
+        /// </summary>
+        /// <param name="quotedText"></param>
+        private static void ValidateQuotedText(string quotedText)
+        {
+            //empty is okay
+
+            for (int i = 0; i < quotedText.Length; i++)
+            {
+                if (IsLinearWhiteSpaceChar(quotedText[i]))
+                    continue;
+
+                if (quotedText[i] <= ' ' || quotedText[i] >= 0xFF)
+                    throw new ArgumentException(SR.Get(SRID.InvalidParameterValue));
+                else
+                    if (quotedText[i] == '"' &&
+                        (i == 0 || quotedText[i - 1] != '\\'))
+                    throw new ArgumentException(SR.Get(SRID.InvalidParameterValue));
+            }
+        }
+
+        /// <summary>
+        /// Returns true if the input character is an allowed character
+        /// Returns false if the input cha----ter is not an allowed character
+        /// </summary>
+        /// <param name="character">input character</param>
+        /// <returns></returns>
+        private static bool IsAllowedCharacter(char character)
+        {
+            //We did not use any of the .Contains methods as
+            //it will result in boxing costs.
```

Comment:
Yes, this certainly could be re-written.  However, I am reluctant to change this - this code is quite old - I took from the original System.IO.Packaging, as is (with a few super-simple changes such as fixing the formatting of the if statement previously referenced).  As much as is possible, I want to retain, at least for now, the majority of the code as is, particularly code that could potentially impact functionality.  I propose we leave the code as-is for now.